app/models/miq_request_task.rb

```diff
@@ -140,12 +140,13 @@ def deliver_queue(req_type = request_type, zone = nil)
 
     _log.info("Queuing #{request_class::TASK_DESCRIPTION}: [#{description}]...")
 
-    workflow = ConfigurationScriptPayload.find(options[:configuration_script_payload_id]) if options[:configuration_script_payload_id]
+    workflow_id = resource_action&.configuration_script_id || options[:configuration_script_payload_id]
```

Comment:
This is a little complicated and I need to do some investigation to see if we can clean this up.

Basically if a Generic ServiceTemplate is created with an Embedded Workflow for an Entry Point that is set in the resource_action.
If a e.g. VMware ServiceTemplate is created the workflow goes to `options[:configuration_script_payload_id]` also when the child tasks are created.